merlin/models/torch/blocks/attention.py

```diff
@@ -0,0 +1,149 @@
+from copy import deepcopy
+from typing import Dict, Optional, Union
+
+import torch
+from torch import nn
+
+from merlin.models.torch.batch import Batch
+from merlin.models.torch.block import Block
+
+
+class CrossAttentionBlock(Block):
+    """
```

Comment:
It would be nice to have some example usage in the docstring because things like having to set `batch_first=True` can only be learned by digging into unit tests.